install_script_op_worker2.sh

```diff
@@ -524,73 +466,16 @@ else
     $sudo_cmd sh -c "echo DD_SITE=$site >> $env_file"
   fi
 
-  # sources
-  if [ "$source_splunk_hec_address" ]; then
-    printf "\033[34m  * Assigning DD_OP_SOURCE_SPLUNK_HEC_ADDRESS.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_SOURCE_SPLUNK_HEC_ADDRESS=$source_splunk_hec_address>> $env_file"
-  fi
-
-  if [ "$source_splunk_tcp_address" ]; then
-    printf "\033[34m  * Assigning DD_OP_SOURCE_SPLUNK_TCP_ADDRESS.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_SOURCE_SPLUNK_TCP_ADDRESS=$source_splunk_tcp_address>> $env_file"
-  fi
-
-  if [ "$source_datadog_agent_address" ]; then
-    printf "\033[34m  * Assigning DD_OP_SOURCE_DATADOG_AGENT_ADDRESS.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_SOURCE_DATADOG_AGENT_ADDRESS=$source_datadog_agent_address>> $env_file"
-  fi
-
-  if [ "$source_sumo_logic_address" ]; then
-    printf "\033[34m  * Assigning DD_OP_SOURCE_SUMO_LOGIC_ADDRESS.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_SOURCE_SUMO_LOGIC_ADDRESS=$source_sumo_logic_address>> $env_file"
-  fi
-
-  # source key passes
-  if [ "$source_splunk_tcp_key_pass" ]; then
-    printf "\033[34m  * Assigning DD_OP_SOURCE_SPLUNK_TCP_KEY_PASS.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_SOURCE_SPLUNK_TCP_KEY_PASS=$source_splunk_tcp_key_pass>> $env_file"
-  fi
-
-  if [ "$source_splunk_hec_key_pass" ]; then
-    printf "\033[34m  * Assigning DD_OP_SOURCE_SPLUNK_HEC_KEY_PASS.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_SOURCE_SPLUNK_HEC_KEY_PASS=$source_splunk_hec_key_pass>> $env_file"
-  fi
-
-  # destinations
-  if [ "$dest_splunk_hec_endpoint" ]; then
-    printf "\033[34m  * Assigning DD_OP_DESTINATION_SPLUNK_HEC_ENDPOINT_URL.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_DESTINATION_SPLUNK_HEC_ENDPOINT_URL=$dest_splunk_hec_endpoint>> $env_file"
-  fi
-
-  if [ "$dest_splunk_hec_token" ]; then
-    printf "\033[34m  * Assigning DD_OP_DESTINATION_SPLUNK_HEC_TOKEN.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_DESTINATION_SPLUNK_HEC_TOKEN=$dest_splunk_hec_token>> $env_file"
-  fi
-
-  if [ "$dest_sumo_logic_endpoint" ]; then
-    printf "\033[34m  * Assigning DD_OP_DESTINATION_SUMO_LOGIC_HTTP_COLLECTOR_URL.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_DESTINATION_SUMO_LOGIC_HTTP_COLLECTOR_URL=$dest_sumo_logic_endpoint>> $env_file"
-  fi
-
-  if [ "$dest_datadog_archives_aws_key_id" ]; then
-    printf "\033[34m  * Assigning DD_OP_DESTINATION_DATADOG_ARCHIVES_AWS_ACCESS_KEY_ID.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_DESTINATION_DATADOG_ARCHIVES_AWS_ACCESS_KEY_ID=$dest_datadog_archives_aws_key_id>> $env_file"
-  fi
-
-  if [ "$dest_datadog_archives_aws_secret" ]; then
-    printf "\033[34m  * Assigning DD_OP_DESTINATION_DATADOG_ARCHIVES_AWS_SECRET_ACCESS_KEY.\n\033[0m\n"
-    $sudo_cmd sh -c "echo DD_OP_DESTINATION_DATADOG_ARCHIVES_AWS_SECRET_ACCESS_KEY=$dest_datadog_archives_aws_secret>> $env_file"
-  fi
+  # All env vars that are defined and used exclusively in OPW, are prefixed with `DD_OP_`
+  for dd_op_var in ${!DD_OP_@}; do
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏What's OPW's behavior if it encounters unknown variables/configuration here?

Reply:
My understanding is that this env file is only used in conjunction with systemd.
OPW is only checking values for the env vars it cares about/knows about.
So if there is any env var that OPW doesn't explicitly read that is written in the env file, it goes ignored.

Reply:
That sounds right from an "outside-OPW" angle - there's no UX concern about someone _trying_ to set a variable that isn't supported and wondering why OPW isn't working as expected?

I assume that's _fine_, but wanted to double check the product/UX side of it.

Reply:
I think I understand the scenario you are depicting. I could see it as a nice-to-have UX. With where we are right now, I don't think it's an issue to not be able to detect when the user is trying to export a DD_OP_ prefixed env var that we don't actually support. FWIW we don't have that functionality today without this PR. I could see it being a handy enhancement though. We could consider flagging that warning from the worker itself, and avoid having to maintain a special public list of the env vars.